imports/plugins/core/accounts/client/containers/addressBookContainer.js

```diff
@@ -0,0 +1,212 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { compose } from "recompose";
+import { Meteor } from "meteor/meteor";
+import { Template } from "meteor/templating";
+import { i18next } from "/client/api";
+import { Countries } from "/client/collections";
+import * as Collections from "/lib/collections";
+import { registerComponent, composeWithTracker } from "@reactioncommerce/reaction-components";
+import AddressBook from "../components/addressBook";
+
+/**
+ * @method updateAddress
+ * @summary helper function that updates an address in the account's addressBook via a meteor method.
+ * @since 2.0.0
+ * @param {Object} address - address to be updated.
+ * @param{String} property - property to be updated.
+ * @return {Promise}
+ */
+function updateAddress(address, property) {
+  return new Promise((resolve, reject) => {
+    Meteor.call("accounts/addressBookUpdate", address, null, property, (error, result) => {
+      if (error || !result) {
+        reject(i18next.t("addressBookGrid.somethingWentWrong", { err: error.message }));
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+/**
+ * @method removeAddress
+ * @summary helper function that updates an address in the account's addressBook via a meteor method.
+ * @since 2.0.0
+ * @param {String} _id - _id of address to be removed.
+ * @return {Promise}
+ */
+function removeAddress(_id) {
+  return new Promise((resolve, reject) => {
+    Meteor.call("accounts/addressBookRemove", _id, (error, result) => {
+      if (error || !result) {
+        reject(i18next.t("addressBookGrid.cantRemoveThisAddress", { err: error.message }));
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+/**
+ * @method addAddress
+ * @summary helper function that adds an address in the account's addressBook via a meteor method.
+ * @since 2.0.0
+ * @param {Object} address - address to be added.
+ * @return {Promise}
+ */
+function addAddress(address) {
+  return new Promise((resolve, reject) => {
+    Meteor.call("accounts/validateAddress", address, (error, result) => {
+      if (error || !result || !result.validated) reject(i18next.t("addressBookAdd.failedToAddAddress", { err: error.message }));
+      Meteor.call("accounts/addressBookAdd", address, (err, res) => {
+        if (err || !res) {
+          reject(i18next.t("addressBookAdd.failedToAddAddress", { err: err.message }));
+        } else {
+          resolve(res);
+        }
+      });
```

Comment:
`reject` does not return out of the function, so for errors in the outer call it will still call `accounts/addressBookAdd` with this logic. You can just add an explicit return:

```js
if (error || !result || !result.validated) {
  reject(i18next.t("addressBookAdd.failedToAddAddress", { err: error.message }));
  return;
}
```